#!/usr/bin/env python
PACKAGE="vimjay"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("height", int_t, 1, "image height", 480, 1, 4096)
gen.add("width", int_t, 1, "image width", 640, 1, 4096)
# TODO(lucasw) make this an enum
gen.add("distortion_model", str_t, 3, "", "plumb_bob")
gen.add("d0", double_t, 3, "distortion coefficient 0", 0, -2.0, 2.0)
gen.add("d1", double_t, 3, "distortion coefficient 1", 0, -2.0, 2.0)
gen.add("d2", double_t, 3, "distortion coefficient 2", 0, -2.0, 2.0)
gen.add("d3", double_t, 3, "distortion coefficient 3", 0, -2.0, 2.0)
gen.add("d4", double_t, 3, "distortion coefficient 4", 0, -2.0, 2.0)
# TODO(lucasw) later may have an entry for every matrix element
gen.add("cx", double_t, 4, "center x", 320.0, -4096, 4096)
gen.add("cy", double_t, 4, "center y", 240.0, -4096, 4096)
gen.add("fx", double_t, 4, "focal x", 400.0, -4096.0, 4096.0)
gen.add("fy", double_t, 4, "focal y", 400.0, -4096.0, 4096.0)
gen.add("update_rate", double_t, 5, "updates per second", 1, 0, 20)

exit(gen.generate(PACKAGE, "dr_camera_info", "CameraInfo"))
